python_action/clang_tidy.py

```diff
@@ -0,0 +1,89 @@
+"""Parse output from clang-tidy's stdout"""
+import os
+import sys
+import re
+from . import GlobalParser
+
+
+class TidyNotification:
+    """Create a object that decodes info from the clang-tidy output's initial line that
+    details a specific notification.
+
+    Attributes:
+        diagnostic (str): The clang-tidy check that enabled the notification.
+        filename (str): The source filename concerning the notification.
+        line (int): The line number of the source file.
+        cols (int): The columns of the line that triggered the notification.
+        note_type (str): The priority level of notification (warning/error).
+        note_info (str): The rationale of the notification.
+        fixit_lines (list): A `list` of lines (`str`) for the code-block in the notification.
+    """
+
+    def __init__(self, notification_line: str):
+        """
+        Args:
+            notification_line: The first line in the notification.
+        """
+        sliced_line = notification_line.split(":")
+        if sys.platform.startswith("win32") and len(sliced_line) > 5:
+            # sliced_list items 0 & 1 are the path seperated at the ":".
+            # we need to re-assemble the path for correct list expansion (see below)
+            sliced_line = [sliced_line[0] + ":" + sliced_line[1]] + sliced_line[2:]
+        (
+            self.filename,
+            self.line,
+            self.cols,
+            self.note_type,
+            self.note_info,
+        ) = sliced_line
+
+        self.diagnostic = re.search("\[.*\]", self.note_info).group(0)
```

Comment:
These backslashes are not anomalous. Both `[` and `]` have a special meaning in regex patterns. In this case the backslashes escapes the special purposes in the regex pattern `\[.*\]`.